build-system/common/ci.js

```diff
@@ -15,121 +15,231 @@
 }
 
 /**
- * Returns the commit SHA being tested by the PR build.
+ * Returns the commit SHA being tested by a PR build.
  * @return {string}
  */
 function ciPullRequestSha() {
-  if (!isPullRequestBuild()) {
-    log(red('ERROR:'), 'Not a PR build, PR SHA is undefined.');
-    return '';
-  }
-  return serviceFunctionMap['ciPullRequestSha']();
+  return isTravis
+    ? env('TRAVIS_PULL_REQUEST_SHA')
+    : isGithubActions
+    ? env('GITHUB_SHA')
+    : isCircleci
+    ? env('CIRCLE_SHA1')
+    : '';
+}
+
+/**
+ * Returns the branch for push builds.
+ * @return {string}
+ */
+function ciPushBranch() {
+  return isTravis
+    ? env('TRAVIS_BRANCH')
+    : isGithubActions
+    ? env('GITHUB_REF') // TODO(rsimha): Truncate if necessary.
+    : isCircleci
+    ? env('CIRCLE_BRANCH')
+    : '';
+}
+
+/**
+ * Returns the commit SHA being tested by a push build.
+ * @return {string}
+ */
+function ciCommitSha() {
+  return isTravis
+    ? env('TRAVIS_COMMIT')
+    : isGithubActions
+    ? env('GITHUB_SHA')
+    : isCircleci
+    ? env('CIRCLE_SHA1')
+    : '';
+}
+
+/**
+ * Returns the number of the current build.
+ * @return {string}
+ */
+function ciBuildNumber() {
+  return isTravis
+    ? env('TRAVIS_BUILD_NUMBER')
+    : isGithubActions
+    ? env('GITHUB_RUN_ID')
+    : isCircleci
+    ? env('CIRCLE_BUILD_NUM')
+    : '';
+}
+
+/**
+ * Returns the URL of the current build.
+ * @return {string}
+ */
+function ciBuildUrl() {
+  return isTravis
+    ? env('TRAVIS_BUILD_WEB_URL')
+    : isGithubActions
+    ? `${env('GITHUB_SERVER_URL')}/${env('GITHUB_REPOSITORY')}/actions/runs/${env('GITHUB_RUN_ID')}` // prettier-ignore
+    : isCircleci
+    ? env('CIRCLE_BUILD_URL')
+    : '';
+}
+
+/**
+ * Returns the number of the current job.
+ * @return {string}
+ */
+function ciJobNumber() {
+  return isTravis
+    ? env('TRAVIS_JOB_NUMBER')
+    : isGithubActions
+    ? env('GITHUB_RUN_NUMBER')
+    : isCircleci
+    ? env('CIRCLE_NODE_INDEX')
+    : '';
+}
+
+/**
+ * Returns the URL of the current job.
+ * @return {string}
+ */
+function ciJobUrl() {
+  return isTravis
+    ? env('TRAVIS_JOB_WEB_URL')
+    : isGithubActions
+    ? `${env('GITHUB_SERVER_URL')}/${env('GITHUB_REPOSITORY')}/runs/${env('GITHUB_RUN_NUMBER')}` // prettier-ignore
+    : isCircleci
+    ? env('CIRCLE_BUILD_URL') // TODO(rsimha): Test and modify if necessary.
+    : '';
+}
+
+/**
+ * Returns the repo slug for the ongoing build.
+ * @return {string}
+ */
+function ciRepoSlug() {
+  return isTravis
+    ? env('TRAVIS_REPO_SLUG')
+    : isGithubActions
+    ? env('GITHUB_REPOSITORY')
+    : isCircleci
+    ? `${env('CIRCLE_PROJECT_USERNAME')}/${env('CIRCLE_PROJECT_REPONAME')}`
+    : '';
 }
 
 module.exports = {
+  ciBuildNumber,
+  ciBuildUrl,
+  ciCommitSha,
+  ciJobNumber,
+  ciJobUrl,
   ciPullRequestBranch,
   ciPullRequestSha,
+  ciPushBranch,
+  ciRepoSlug,
   isCiBuild,
+  isCircleciBuild,
+  isGithubActionsBuild,
   isPullRequestBuild,
   isPushBuild,
+  isTravisBuild,
```

Comment:
Question: Is there a need for these to be callable functions, or would exporting module-level constants a la `IS_TRAVIS` (like are already defined above) be sufficient? AFAIK the `process.env` values shouldn't change at runtime

Reply:
Taking a second pass look, all of these seem to be process-constant values. Do you see downsides to exporting a plain `Map`/`dict` with each of these properties (not unline `process.env` itself)?

Reply:
I can see how that would be a concern. FWIW, there's a total of 33 call sites for all of these combined (most of which are being touched in this PR). While there are things that could be dynamic, many of these properties like whether the process in running on Travis or GitHub Actions, on PR or push, etc. are about a constant as it gets w.r.. I feel the added simplicity is worth consideration

Reply:
There are hundreds of call sites that use `ci.js`. In any case, the cost of a JS function call to access a property is near zero, and it's worth holding on to the flexibility / readability advantages of the code as written.

Reply:
To clarify scale in the event of future refactor considerations, there are no longer hundreds of call sites using `ci.js` (perhaps many have already been removed in previous refactors?). `ci` is `require`'d in under 40 files, including `ci` itself.

Reply:
There are ~40 files and ~200 call-sites.

Reply:
I can see how you arrived at those numbers, as an initial search for this set of functions at this commit turns up 213 matches across 46 files. I would have been in the same boat, until noticing that around two thirds of those matches are either in the require statements themselves, `ci.js`, or the `.cache` directory of the `@babel` node module. The number of actual call-sites is well below 100, making a future refactor much more practical than it may seem at first

Reply:
I stand corrected on the numbers. The main reasons behind not wanting to go down this path (second sentence in https://github.com/ampproject/amphtml/pull/31527#discussion_r540382585) still remain.